fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/plans/RecordQueryInUnionPlan.java

```diff
@@ -0,0 +1,419 @@
+    }
+
+    @Nonnull
+    @Override
+    public <M extends Message> RecordCursor<QueryResult> executePlan(@Nonnull final FDBRecordStoreBase<M> store,
+                                                                     @Nonnull final EvaluationContext context,
+                                                                     @Nullable final byte[] continuation,
+                                                                     @Nonnull final ExecuteProperties executeProperties) {
+        int size = getValuesSize(context);
+        if (size > maxNumberOfValuesAllowed) {
+            throw new RecordCoreException("too many IN values").addLogInfo("size", size);
+        }
+        final RecordQueryPlan childPlan = getInnerPlan();
+        final ExecuteProperties childExecuteProperties;
+        // Can pass the limit down to all sides, since that is the most we'll take total.
+        if (executeProperties.getSkip() > 0) {
+            childExecuteProperties = executeProperties.clearSkipAndAdjustLimit();
+        } else {
+            childExecuteProperties = executeProperties;
+        }
+        final List<Function<byte[], RecordCursor<FDBQueriedRecord<M>>>> childCursorFunctions = getValuesContexts(context).stream()
+                .map(childContext -> (Function<byte[], RecordCursor<FDBQueriedRecord<M>>>)childContinuation -> childPlan.execute(store, childContext, childContinuation, childExecuteProperties))
+                .collect(Collectors.toList());
+        return UnionCursor.create(store, comparisonKey, reverse, childCursorFunctions, continuation)
+                .skipThenLimit(executeProperties.getSkip(), executeProperties.getReturnedRowLimit())
+                .map(QueryResult::of);
+    }
+
+    @Nonnull
+    public RecordQueryPlan getInnerPlan() {
+        return inner.getRangesOverPlan();
+    }
+
+    @Override
+    @Nonnull
+    public RecordQueryPlan getChild() {
+        return getInnerPlan();
+    }
+
+    @Override
+    public boolean isReverse() {
+        return reverse;
+    }
+
+    @Override
+    public void logPlanStructure(final StoreTimer timer) {
+        timer.increment(FDBStoreTimer.Counts.PLAN_IN_UNION);
+        getInnerPlan().logPlanStructure(timer);
+    }
+
+    @Nonnull
+    @Override
+    public List<? extends Value> getResultValues() {
+        return resultValuesSupplier.get();
+    }
+
+    @Nonnull
+    @Override
+    public PlannerGraph rewritePlannerGraph(@Nonnull final List<? extends PlannerGraph> childGraphs) {
+        return PlannerGraph.fromNodeAndChildGraphs(
+                new PlannerGraph.OperatorNodeWithInfo(this,
+                        NodeInfo.IN_UNION_OPERATOR,
+                        valuesSources.stream().map(Object::toString).collect(Collectors.toList()),
+                        ImmutableMap.of()),
+                childGraphs);
+    }
+
+    @Nonnull
+    @Override
+    public Set<CorrelationIdentifier> getCorrelatedToWithoutChildren() {
+        return ImmutableSet.of();
```

Comment:
It's okay to leave as is but this is not correct I think. won't trip up anybody right now since that logic is only used in the new planner.